### ISSUE TITLE
plot from hs.interactive not updating with events

### ISSUE THREAD

**Issue by thomasaarholt** [hyperspy/hyperspy#1988]
From the [docs](http://hyperspy.org/hyperspy-doc/current/user_guide/tools.html#region-of-interest-roi):
```
%matplotlib notebook
import hyperspy.api as hs
import scipy.misc
im = hs.signals.Signal2D(scipy.misc.ascent())
im.plot()
roi = hs.roi.RectangularROI(left=30, right=500, top=200, bottom=400)
im_roi = roi.interactive(im, color="red")
roi_hist =hs.interactive(im_roi.get_histogram,
                         event=im_roi.axes_manager.events.\
                               any_axis_changed,
                         recompute_out_event=None)
roi_hist.plot()
```
This should show the following behaviour:
![image_with_rect_roi](https://user-images.githubusercontent.com/2721423/42125195-3d7faa1a-7c72-11e8-911e-cf2973ff5d11.gif)

However, dragging around does not change the plot at all. 

Manually calling roi_hist.plot() after moving the ROI shows a changed plot, so data in `roi_hist`is being updated, but is not being shown on the plot.

**Comment by thomasaarholt** [comment]
As a sanity check, this behaviour exists in qt4 as well as the notebook and ipympl backends.

It would be nice to add a test to cover this case, although I'm not sure how that would be done.

**Comment by francisco-dlp** [comment]
Fixed in #1994
